view/src/components/surplus.js

```diff
@@ -0,0 +1,556 @@
+            >
+              <CloseIcon />
+            </IconButton>
+          ) : null}
+        </MuiDialogTitle>
+      );
+    });
+
+    const DialogContent = withStyles((theme) => ({
+      viewRoot: {
+        padding: theme.spacing(2),
+      },
+    }))(MuiDialogContent);
+
+    dayjs.extend(relativeTime);
+    const { classes } = this.props;
+    const { open, errors, viewOpen } = this.state;
+
+    /** Set all states to generic value when opening a dialog page */
+    const handleClickOpen = () => {
+      this.setState({
+        // surplus state
+        surplusId: "",
+        produceId: "",
+        produceName: "",
+        originFarmId: "",
+        originFarmName: "",
+        originFarmLocation: "",
+        originFarmContactName: "",
+        originFarmContactPhone: "",
+        available: false,
+        cost: "",
+        totalQuantityAvailable: "",
+        packagingType: "",
+        // page state
+        buttonType: "",
+        open: true,
+      });
+    };
+
+    const handleViewClose = () => {
+      // page state
+      this.setState({ viewOpen: false });
+    };
+
+    const handleDialogClose = (event) => {
+      // page state
+      this.setState({ open: false });
+    };
+
+    if (this.state.uiLoading === true) {
+      return (
+        <main className={classes.content}>
+          {this.state.uiLoading && <CardSkeletons classes={classes} />}
+        </main>
+      );
+    } else {
+      return (
+        <main className={classes.content}>
+          <div className={classes.toolbar} />
+          <Fab
+            color="primary"
+            className={classes.floatingButton}
+            aria-label="Add Produce"
+            onClick={handleClickOpen}
+          >
+            <AddIcon />
+          </Fab>
+          <Dialog
+            fullScreen
+            open={open}
+            onClose={handleDialogClose}
+            TransitionComponent={Transition}
+          >
+            <AppBar className={classes.appBar}>
+              <Toolbar>
+                <IconButton
+                  edge="start"
+                  color="inherit"
+                  onClick={handleDialogClose}
+                  aria-label="close"
+                >
+                  <CloseIcon />
+                </IconButton>
+                <Typography variant="h6" className={classes.title}>
+                  {this.state.buttonType === "Edit"
+                    ? "Edit Surplus"
+                    : "Create a new Surplus"}
+                </Typography>
+              </Toolbar>
+            </AppBar>
+            <Container maxWidth="lg">
+              <SurplusStepper
+                buttonType={this.state.buttonType}
+                alert={this.props.alert}
+                farmId={this.state.originFarmId}
+                produceId={this.state.produceId}
+                surplusId={this.state.surplusId}
+                handleChange={this.handleChange}
+                handleAsyncChange={this.handleAsyncChange}
```

Comment:
It might be clearer to just inline these rather than defining a separate method for each.  Up to you though, I don't know React that well. :)

Reply:
what do you mean by inline?